hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/service/DirectoryDeletingService.java

```diff
@@ -135,10 +142,65 @@ public void resume() {
   @Override
   public BackgroundTaskQueue getTasks() {
     BackgroundTaskQueue queue = new BackgroundTaskQueue();
-    queue.add(new DirectoryDeletingService.DirDeletingTask(this));
+    supplier.reInitItr();
+    for (int i = 0; i < DIR_DELETING_CORE_POOL_SIZE; i++) {
+      queue.add(new DirectoryDeletingService.DirDeletingTask(this));
+    }
     return queue;
   }
 
+  @Override
+  public void shutdown() {
+    super.shutdown();
+    supplier.closeItr();
+  }
+
+  private final class Supplier {
```

Comment:
rename class to avoid conflict with java.util.function.Supplier, DeletedDirSupplier

Reply:
Done